lxd/instance/drivers/qmp/monitor.go

```diff
@@ -183,9 +183,20 @@ func Connect(path string, serialCharDev string, eventHandler func(name string, d
 		return nil, err
 	}
 
-	err = qmpConn.Connect()
-	if err != nil {
-		return nil, err
+	chError := make(chan error, 1)
+	go func() {
+		err = qmpConn.Connect()
+		chError <- err
+	}()
+
+	select {
+		case err := <-chError:
+			if err != nil {
+				return nil, err
+			}
+		case <-time.After(5 * time.Second):
+			qmpConn.Disconnect()
```

Comment:
@stgraber will this cause the `Connect()` call above to end, cleaning up the go routine?

Reply:
Confirmed that go routine is released on call.

Reply:
Yep, I confirmed that it does.